main.go

```diff
@@ -176,14 +179,33 @@ func main() {
 		if err != nil {
 			log.Fatalf("Failed to marshal the replication controller: %v", err)
 		}
-
-		// Save the replication controller for the Docker compose service to the
-		// configs directory.
-		outputFileName := fmt.Sprintf("%s-rc.json", name)
-		outputFilePath := filepath.Join(outputDir, outputFileName)
-		if err := ioutil.WriteFile(outputFilePath, data, 0644); err != nil {
-			log.Fatalf("Failed to write replication controller %s: %v", outputFileName, err)
+		if !asYml {
+			// Save the replication controller for the Docker compose service to the
+			// configs directory.
+			outputFileName := fmt.Sprintf("%s-rc.json", name)
+			outputFilePath := filepath.Join(outputDir, outputFileName)
+			if err := ioutil.WriteFile(outputFilePath, data, 0644); err != nil {
+				log.Fatalf("Failed to write replication controller %s: %v", outputFileName, err)
+			}
+			fmt.Println(outputFilePath)
+		} else {
+			// Save the replication controller to Yaml file
+			var exp interface{}
+			// because yaml is not directly usable grom api, we can unmarshal json to interface{}
+			// and then write yaml file
+			json.Unmarshal(data, &exp)
```

Comment:
Is there a reason we need to unmarshal the generated json to an interface{} rather than just yaml.Marshal(rc)?